services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -148,33 +174,34 @@ spec = do
           context "no zuser" $ do
             it "responds with 'client error'" $ do
               env <- ask
-              (_, _, idp) <- createTestIdP
-              whichone (env ^. teSpar) Nothing idp
+              let idpid = env ^. teIdP . idpId
+              whichone (env ^. teSpar) Nothing idpid
                 `shouldRespondWith` checkErr (== 400) "client-error"
 
           context "zuser has no team" $ do
             it "responds with 'no team member'" $ do
               env <- ask
-              (_, _, idp) <- createTestIdP
+              let idpid = env ^. teIdP . idpId
               (uid, _) <- call $ createRandomPhoneUser (env ^. teBrig)
-              whichone (env ^. teSpar) (Just uid) idp
+              whichone (env ^. teSpar) (Just uid) idpid
                 `shouldRespondWith` checkErr (== 403) "no-team-member"
 
           context "zuser has wrong team" $ do
             it "responds with 'no team member'" $ do
               env <- ask
-              (_, _, idp) <- createTestIdP
+              let idpid = env ^. teIdP . idpId
               (uid, _) <- call $ createUserWithTeam (env ^. teBrig) (env ^. teGalley)
-              whichone (env ^. teSpar) (Just uid) idp
+              whichone (env ^. teSpar) (Just uid) idpid
                 `shouldRespondWith` checkErr (== 403) "no-team-member"
 
           context "zuser is a team member, but not a team owner" $ do
             it "responds with 'insufficient-permissions' and a helpful message" $ do
               env <- ask
-              (_owner, tid, idp) <- createTestIdP
+              let idpid = env ^. teIdP . idpId
+                  teamid = env ^. teTeamId
               newmember <- let Just perms = newPermissions mempty mempty
-                        in call $ createTeamMember (env ^. teBrig) (env ^. teGalley) tid perms
-              whichone (env ^. teSpar) (Just newmember) idp
+                        in call $ createTeamMember (env ^. teBrig) (env ^. teGalley) teamid perms
```

Comment:
given how often `tid` is used in other services as a shorthand for a variable with type `TeamId`, I'd say that it's better to keep it as `tid`

Reply:
i disagree with this, too.  i think `tid` is almost as bad as the one-letter local variables used in many places.

if you insist, i'll change it, though.